src/app/upload/upload.service.ts

```diff
@@ -33,6 +34,7 @@ export class UploadService {
 
     await Promise.all(promises);
 
+    this.gamification.reward(Rewards.SCAN_UPLOAD);
```

Comment:
This method should return a promise or something, we can't just ignore it because if the user closes the window before the reward is done it will never happen

Reply:
added await